db/postgres/connection.go

```diff
@@ -383,7 +416,7 @@ func (d *DB) Exec(statement string, args ...interface{}) error {
 // if the transaction is already started the same will be returned.
 func (d *DB) BeginTransaction() (connection.DB, error) {
 	if d.tx != nil {
-		return d, nil
+		return nil, errors.New("transaction already exists")
```

Comment:
please make this into an error in the errors lib (this facilitates error checking)